src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -779,29 +779,23 @@ private static string RetrieveProcessUserName(Process process)
 
                     var tokenUser = Marshal.PtrToStructure<Win32Native.TOKEN_USER>(tokenUserInfo);
 
-                    // Set the default length to be 256, so it will be sufficient for most cases.
-                    int userNameLength = 256, domainNameLength = 256;
-                    var userNameStr = new StringBuilder(userNameLength);
-                    var domainNameStr = new StringBuilder(domainNameLength);
+                    // Max username is defined as UNLEN = 256 in lmcons.h
+                    // Max domainname is defined as DNLEN = CNLEN = 15 in lmcons.h
+                    // The buffer length must be +1, last position is for a null string terminator.
+                    int userNameLength = 257;
+                    int domainNameLength = 16;
+                    Span<char> userNameStr = stackalloc char[userNameLength];
+                    Span<char> domainNameStr = stackalloc char[domainNameLength];
                     Win32Native.SID_NAME_USE accountType;
 
                     if (!Win32Native.LookupAccountSid(null, tokenUser.User.Sid, userNameStr, ref userNameLength, domainNameStr, ref domainNameLength, out accountType))
                     {
-                        error = Marshal.GetLastWin32Error();
-                        if (error == Win32Native.ERROR_INSUFFICIENT_BUFFER)
-                        {
-                            userNameStr.EnsureCapacity(userNameLength);
-                            domainNameStr.EnsureCapacity(domainNameLength);
-
-                            if (!Win32Native.LookupAccountSid(null, tokenUser.User.Sid, userNameStr, ref userNameLength, domainNameStr, ref domainNameLength, out accountType)) { break; }
-                        }
-                        else
-                        {
-                            break;
-                        }
+                        break;
                     }
 
-                    userName = domainNameStr + "\\" + userNameStr;
+                    // TODO: Use Concat() from .Net Core 3.0
+                    // public static string Concat(ReadOnlySpan<char> str0, ReadOnlySpan<char> str1, ReadOnlySpan<char> str2)
+                    userName = string.Concat(domainNameStr.Slice(0, domainNameLength).ToString(), "\\", userNameStr.Slice(0, userNameLength).ToString());
```

Comment:
> Slice(0, domainNameLength)

According to the doc about [`LookupAccountSidA`](https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-lookupaccountsida), `userNameLength` and `domainNameLength` will be set to the required buffer size only if the provided buffer is too small.
It doesn't say those two will be set to the actual length of the names when the buffer is large enough. In that case, you don't know how long the actual domain name is, and this will result in a string with trailing `null` characters (`[char]0`).

Have you verified that `userNameLength` and `domainNameLength` will be set to the actual length of the names when the buffers are good and the call succeeds?

Reply:
I can confirm that the LookupAccountSidA returns right lengths 

![image](https://user-images.githubusercontent.com/22290914/54971153-fa0f8f80-4fa6-11e9-9569-733892a0253d.png)